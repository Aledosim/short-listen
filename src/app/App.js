import React from 'react';
import style, { createGlobalStyle } from 'styled-components';

import TopBar from '../features/topbar/TopBar'
import TrackList from '../features/tracklist/TrackList'

// colors
const background = "#0A0A0A"

const GlobalStyle = createGlobalStyle`
	* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
	}

  body {
    background-color: ${background};
  }

	@media(max-width: 1080px) {
    html {
      font-size: 93.75%;
    }
	}

	@media(max-width: 720px) {
    html {
      font-size: 87.5%;
    }
	}

  button {
    cursor: pointer;
  }
`

const Header = style.header`
  position: fixed;
  width: 100%;
  z-index: 1;

  background-color: ${background};
`

const TrackListContainer = style.div`
  position: relative;
  top: 4.5rem;
`

function App() {
  return (
    <>
      <GlobalStyle />
      <Header>
        <TopBar />
      </Header>
      <TrackListContainer>
        <TrackList />
      </TrackListContainer>
    </>
  );
}

export default App;
