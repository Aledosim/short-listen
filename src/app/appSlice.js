import { createSlice } from '@reduxjs/toolkit'

const initialState = {
  value: 'trackList',
}

export const appSlice = createSlice({
  name: 'app',
  initialState,
  reducers: {
    changedToFavorites: (state) => {
      console.log('mudando para favoritos')
      state.value = 'favorites'},
  },
})

export const { changedToFavorites } = appSlice.actions

export const selectView = (state) => state.app.value

export default appSlice.reducer
