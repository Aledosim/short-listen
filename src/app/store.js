import { configureStore } from '@reduxjs/toolkit';

import trackListReducer from '../features/tracklist/trackListSlice'
import favoritesReducer from '../features/favorites/favoritesSlice'

export const store = configureStore({
  reducer: {
    trackList: trackListReducer,
    favorites: favoritesReducer,
  },
});
