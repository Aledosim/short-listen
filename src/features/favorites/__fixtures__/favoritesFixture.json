[
  1391349252,
  1450292592,
  1243010562,
  1451144022,
  1442170372,
  1391349282,
  1397354732,
  1404805642,
  1397165102,
  1391338992
]
