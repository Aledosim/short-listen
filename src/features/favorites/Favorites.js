import React from 'react'
import styled from 'styled-components'
import style from 'styled-components'
import InfiniteScroll from 'react-infinite-scroll-component'
import { useDispatch, useSelector } from 'react-redux'

import { selectFavorites } from './favoritesSlice'
import { selectTrackList } from '../tracklist/trackListSlice'
import TrackCard from '../trackcard/TrackCard'
import { changedToTrackList } from '../../app/appSlice'

import chartIcon from './chart.png'

// colors
const text = "#EBEBEB"

const StyledInfiniteScroll = styled(InfiniteScroll)`
  display: flex;
  flex-direction: column;
  align-items: center;

  color: ${text};
`

const Header = style.header`
  display: flex;
  justify-content: flex-end;

  height: 2.5rem;
  width: 100%;
  z-index: 1;

  position: fixed;
`
const Button = style.button`
  height: 3rem;
  width: 4rem;

  margin-top: 1rem;
  margin-right: 0.5rem;

	background-color: #0A0A0A;
  background-repeat: no-repeat;
  background-position: center;
  background-image: url(${chartIcon});

	border: none;
`

export default function Favorites(){

  const favorites = useSelector(selectFavorites)
  const trackList = useSelector(selectTrackList)
  const dispatch = useDispatch()

  return(
    <>
      <Header>
        <Button
          name='track list'
          onClick={() => dispatch(changedToTrackList())}
        />
      </Header>
      <div role="list">
        <StyledInfiniteScroll
          hasMore={false}
          dataLength={favorites.length}
          loader={<h4>Loading...</h4>}
          endMessage={
            <p style={{ textAlign: 'center' }}>
            <b>Yay! You have seen it all</b>
            </p>
          }
          >
          {
            favorites.map( (trackId, i) => {
              const track = trackList.find((track) => track.id === trackId)
              return(
                <TrackCard
                cover={track.cover}
                title={track.title}
                singer={track.singer}
                time={track.time}
                preview={track.preview}
                link={track.link}
                id={track.id}
                key={i}
                />
              )
            })
          }
        </StyledInfiniteScroll>
      </div>
    </>
  );
};

