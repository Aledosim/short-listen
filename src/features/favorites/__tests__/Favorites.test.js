import React from 'react'
import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'

import { store } from '../../../app/store'
import Favorites from '../Favorites'
import * as favoritesFixture from '../__fixtures__/favoritesFixture.json'
import * as reactreduxModule from 'react-redux'

function testRender() {
    return render(
      <Provider store={store}>
        <Favorites />
      </Provider>
    )
}

jest.mock("../../trackcard/TrackCard", () => {

  function DummyComponent(props) {

    const componentProps = Object.keys(props).map(value => {
      return(
        <div
          key={value}
          data-testid={value}
        >
          {props[value]}
        </div>
      )
    })

    return(
      <div role="listitem">
        {componentProps}
      </div>
    )
  }

  return {
    __esModule: true,
    default: DummyComponent
  }
})

describe('<Favorites /> tests', () => {

  it('renders without crashing', async () => {
    const tree = testRender()

    expect(tree).toMatchSnapshot()
  })

  it('should contain a list of tracks', async () => {
    jest.spyOn(reactreduxModule, 'useSelector')
      .mockImplementation(() => favoritesFixture.default)

    testRender()

    expect(screen.getByRole("list"))
    expect(screen.getAllByRole("listitem"))
  })

})

