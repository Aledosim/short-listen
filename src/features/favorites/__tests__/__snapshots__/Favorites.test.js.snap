// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites /> tests renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #EBEBEB;
}

<body>
    <div>
      <header>
        <button
          name="track list"
        />
      </header>
      <div
        role="list"
      >
        <div
          class="infinite-scroll-component__outerdiv"
        >
          <div
            class="infinite-scroll-component c0"
            style="height: auto; overflow: auto;"
          >
            <p
              style="text-align: center;"
            >
              <b>
                Yay! You have seen it all
              </b>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <button
        name="track list"
      />
    </header>
    <div
      role="list"
    >
      <div
        class="infinite-scroll-component__outerdiv"
      >
        <div
          class="infinite-scroll-component sc-bdnxRM kyjttw"
          style="height: auto; overflow: auto;"
        >
          <p
            style="text-align: center;"
          >
            <b>
              Yay! You have seen it all
            </b>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
